EngineeringModel/ViewModels/ElementDefinitionBrowser/ElementDefinitionTreeRows/ParameterValueBaseRowViewModel.cs

```diff
@@ -524,11 +536,25 @@ private void AddValueSetListener(ParameterValueSetBase thing)
                 return;
             }
 
-            var listener = CDPMessageBus.Current.Listen<ObjectChangedEvent>(thing)
-                            .Where(objectChange => objectChange.EventKind == EventKind.Updated && objectChange.ChangedThing.RevisionNumber > this.RevisionNumber)
-                            .ObserveOn(RxApp.MainThreadScheduler)
-                            .Subscribe(_ => this.SetValues());
-            this.Disposables.Add(listener);
+            Func<ObjectChangedEvent, bool> discriminator = objectChange => objectChange.EventKind == EventKind.Updated && objectChange.ChangedThing.RevisionNumber > this.RevisionNumber;
+            Action<ObjectChangedEvent> action = x => this.SetValues();
+
+            if (this.AllowMessageBusSubscriptions)
```

Comment:
will this code ever execute? i doubt it will since you've refactored the whole ElementDef browser and this class is only used there... if that is the case, should we not remove?

Reply:
I think that is the same as for others. 
I initially chose to do it like this, so no other code (unexpected usages of ElementDefinitionTree RowViewModels) will fail. 
I had some unexpected failing unit tests before I created the automatic setting of the AllowMessageBusSubscriptions property in ViewModelBase. Before, I set the AllowMessageBusSubscriptions property "manually" in the base rowviewmodel classes. Some of these RowViewModels are definitely used outside the scope of the ElementDefinitionBrowser.